code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -58,20 +56,35 @@
 		M.adjust_fire_stacks(reac_volume / 15)
 	return ..()
 
+/datum/reagent/consumable/ethanol/on_mob_delete(mob/living/M)
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		spawn(300)
+			if(H)
+				H.drunk_state = NOT_DRUNK
```

Comment:
When the alcohol reagent leaves someone's body, the drunk state doesn't ever reset. I'll optimize it in the next commit.